backend/crates/eiffelvis_core/src/domain/user_queries.rs

```diff
@@ -148,17 +148,18 @@ impl<I> TrackedQuery<I> {
             if self.event_filters.is_empty() {
                 true
             } else {
+                // Event filters itterator that returns nodes that contain the imput given (id filter has to be an exact match, all others do not.)
                 self.event_filters.iter().filter(|v| !v.is_empty()).any(|filters| {
                     filters.iter().all(|filter| match &filter.pred {
-                        EventFilter::Type { names: ref name } => name.iter().any(|name| &node.data().meta.event_type == name),
+                        EventFilter::Type { names: ref name } => name.iter().any(|name| node.data().meta.event_type.to_lowercase().contains(&name.to_lowercase())),
```

Comment:
`to_lowercase` allocates a new string on every invocation, cache the value for `node` at least.
But also reconsider if you really need it to be case insensitive to prevent it all together.

Reply:
Hi Ruthger, 

Not sure if I understand you correctly but you want the node (with attributes converted to lowercase) to be cached then the comparison is done with the cached node? 

Regarding the case  insensitivity, not having this defeats the point of this as it doesn't allow as much flexibility. It would be irritating to have to remember/check the case of the event info you want to filter, if you need to check you could just copy the exact name and filter that way.

Reply:
Regarding caching, I mean that node.data() is called every time even though `node` is immutable in that loop hence can be moved out outside, to line 151